node/src/actors/epoch_manager/mod.rs

```diff
@@ -126,6 +129,13 @@ impl EpochManager {
                 // Start checkpoint monitoring process
                 actor.checkpoint_monitor(ctx);
 
+                // Start ntp update process
+                let mut global_ts = get_global_timestamp();
+                global_ts.server = config.ntp.server.clone();
+                set_global_timestamp(global_ts);
```

Comment:
There is a race condition here because the global timestamp could change between the call to get and the call to set (if a different thread calls set).

This set should only modify global_ts.server, not the entire structure. So instead of `set`, a better API would be `get_mut`.